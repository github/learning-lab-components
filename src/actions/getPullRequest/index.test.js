const getPullRequest = require('.')
const mockContext = require('../../../tests/mockContext')

describe('getPullRequest', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      pull_request: {
        number: 1
      }
    }, {
      pullRequests: {
        get: jest.fn(() => Promise.resolve({ data: { number: 1 } }))
      },
      search: {
        issues: jest.fn(() => Promise.resolve({
          data: {
            items: [{ title: 'My Pull Request', number: 1 }]
          }
        }))
      }
    })
  })

  it('gets a pull request', async () => {
    const pr = await getPullRequest(context, {})
    expect(pr).toMatchSnapshot()
  })

  it('gets a pull request with a title string', async () => {
    const pr = await getPullRequest(context, { pullRequest: 'My Pull Request' })
    expect(pr).toMatchSnapshot()
  })

  it('gets a pull request with a pr number', async () => {
    const pr = await getPullRequest(context, { pullRequest: 1 })
    expect(pr).toMatchSnapshot()
  })
})
