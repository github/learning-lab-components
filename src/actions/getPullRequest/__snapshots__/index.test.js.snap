// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPullRequest gets a pull request 1`] = `
Object {
  "data": Object {
    "number": 1,
  },
}
`;

exports[`getPullRequest gets a pull request with a pr number 1`] = `
Object {
  "data": Object {
    "number": 1,
  },
}
`;

exports[`getPullRequest gets a pull request with a title string 1`] = `
Object {
  "data": Object {
    "number": 1,
  },
}
`;
